types/applications/dialog.d.ts

```diff
@@ -0,0 +1,330 @@
+   *
+   * @param title       - The confirmation window title
+   * @param content     - The confirmation message
+   * @param yes         - Callback function upon yes
+   * @param no          - Callback function upon no
+   * @param render      - A function to call when the dialog is rendered
+   * @param defaultYes  - Make "yes" the default choice?
+   *                      (default: `true`)
+   * @param rejectClose - Reject the Promise if the Dialog is closed without making a choice.
+   *                      (default: `false`)
+   * @param options     - Additional rendering options passed to the Dialog
+   *                      (default: `{}`)
+   *
+   * @returns A promise which resolves once the user makes a choice or closes the window
+   *
+   * @example
+   * ```typescript
+   * let d = Dialog.confirm({
+   *  title: "A Yes or No Question",
+   *  content: "<p>Choose wisely.</p>",
+   *  yes: () => console.log("You chose ... wisely"),
+   *  no: () => console.log("You chose ... poorly"),
+   *  defaultYes: false
+   * });
+   * ```
+   */
+  static confirm<Y = true, N = false, R extends boolean = false>({
+    title,
+    content,
+    yes,
+    no,
+    render,
+    defaultYes,
+    rejectClose,
+    options
+  }?: {
+    title: string;
+    content: string;
+    yes?: (html: JQuery) => Y;
+    no?: (html: JQuery) => N;
+    render: Function;
+    defaultYes?: boolean;
+    rejectClose: R;
+    options?: Partial<Dialog.Options> & { jQuery: true };
+  }): Promise<R extends true ? Y | N : Y | N | null>;
+  static confirm<Y = true, N = false, R extends boolean = false>({
+    title,
+    content,
+    yes,
+    no,
+    render,
+    defaultYes,
+    rejectClose,
+    options
+  }?: {
+    title: string;
+    content: string;
+    yes?: (html: HTMLElement) => Y;
+    no?: (html: HTMLElement) => N;
+    render: Function;
+    defaultYes?: boolean;
+    rejectClose: R;
+    options?: Partial<Dialog.Options> & { jQuery: false };
+  }): Promise<R extends true ? Y | N : Y | N | null>;
+  static confirm({
+    title,
+    content,
+    yes,
+    no,
+    render,
+    defaultYes,
+    rejectClose,
+    options
+  }?: {
+    title: string;
+    content: string;
+    yes?: Function;
+    no?: Function;
+    render: Function;
+    defaultYes?: boolean;
+    rejectClose?: boolean;
+    options?: Partial<Dialog.Options>;
+  }): Promise<any>;
+  /**
+   * @deprecated You are passing an options object as a second parameter to Dialog.confirm. This should now be passed in
+   *             as the options key of the first parameter.
+   */
+  static confirm(
+    {
+      title,
+      content,
+      yes,
+      no,
+      render,
+      defaultYes,
+      rejectClose,
+      options
+    }?: {
+      title: string;
+      content: string;
+      yes?: Function;
+      no?: Function;
+      render: Function;
+      defaultYes?: boolean;
+      rejectClose?: boolean;
+      options?: Partial<Dialog.Options>;
+    },
+    old?: Partial<Dialog.Options>
```

Comment:
I think `old` should not be optional for this overload. In general, the type definitions for this overload could probably be improved, similar to what you did with the other overloads, but I don't think it's worth it since it is deprecated.

Reply:
It has to be optional, because the first parameter is already optional. That's kinda nonsensical, as it has required members, but that's foundry.